pkg/ipmanager/ipam/ippool.go

```diff
@@ -0,0 +1,115 @@
+package ipam
+
+import (
+	"encoding/binary"
+	"math"
+	"net"
+	"sync"
+)
+
+type IpPool struct {
+	cidr string
+	net  *net.IPNet
+	size int
+	available map[string]bool //IP.String() is key
+	allocated map[string]string //resource name is key, ipString is value
+	sync.RWMutex
+}
+
+func NewIpPool(cidr string) (*IpPool, error) {
```

Comment:
We have to handle a restart scenario as well. In such cases, as we discussed, we will have to reconstruct/re-initialize this pool from the globalIPs allocated to the pods/services.